src/base/static/js/views/geometry-editor-view.js

```diff
@@ -0,0 +1,583 @@
+      this.workingGeometry.disable();
+      this.workingGeometry = null;
+      this.geometry = null;
+    }
+  },
+
+  updateColorpicker: function() {
+    $(".toolbar-colorpicker-container").spectrum("set",
+      tinycolor(this.colorpickerSettings[this.DRAWING_DEFAULTS[this.colorpickerSettings.editMode].color])
+        .setAlpha(this.colorpickerSettings[this.DRAWING_DEFAULTS[this.colorpickerSettings.editMode].opacity]).toRgbString());
+  },
+
+  saveWorkingGeometry: function() {
+    if (this.workingGeometry) {
+      this.workingGeometry.save();
+    }
+  },
+
+  displayHelpMessage: function(msg) {
+    this.$el.find(".helper-messages ." + msg)
+      .removeClass("hidden")
+      .siblings()
+      .addClass("hidden");
+  },
+
+  showIconToolbar: function() {
+    this.$el.find(".geometry-toolbar-icon-field")
+      .removeClass("hidden");
+  },
+
+  setIconToolbarIcon: function() {
+    this.$el.find(".geometry-toolbar-icon-field :input[value='" + this.iconUrl + "']")
+      .prop("checked", true);
+  },
+
+  hideIconToolbar: function() {
+    this.$el.find(".geometry-toolbar-icon-field").addClass("hidden");
+  },
+
+  setGeometryToolbarHighlighting: function(currentTarget) {
+    var target = this.$el.find(currentTarget)
+
+    if (target.hasClass("selected")) {
+      target.removeClass("selected");
+    } else {
+      target
+        .addClass("selected")
+        .siblings()
+        .removeClass("selected");
+    }
+  },
+
+  swapToolbarVisibility: function() {
+    this.$geometryToolbarEdit.toggleClass("hidden");
+    this.$geometryToolbarCreate.toggleClass("hidden");
+
+    if (this.geometryType === "Point" ||
+        this.geometryType === "marker") {
+
+      this.$geometryToolbarEdit.find(".colorpicker").addClass("hidden");
+    } else {
+      this.$geometryToolbarEdit.find(".colorpicker").removeClass("hidden");
+    }
+
+    this.resetGeometryToolbarHighlighting();
+  },
+
+  resetGeometryToolbarHighlighting: function() {
+    this.$el.find(".geometry-toolbar button")
+      .removeClass("selected");
+  },
+
+  addLayerToMap: function(layer) {
+    if (!this.map.hasLayer(layer)) {
+      this.map.addLayer(layer);
+    }
+  },
+
+  removeLayerFromMap: function(layer) {
+    if (layer) {
+      this.map.removeLayer(layer);
+    }
+  },
+
+  changeLayerGroup: function(layer, sourceLayerGroup, targetLayerGroup) {
+    sourceLayerGroup.removeLayer(layer);
+    targetLayerGroup.addLayer(layer);
+  },
+
+  setEditingCursor: function() {
+    $(".leaflet-container").addClass("add-edit-cursor");
+  },
+
+  setDefaultCursor: function() {
+    $(".leaflet-container").removeClass("add-edit-cursor");
+  },
+
+  getLayerFromEditingLayerGroup: function() {
+    var returnLayer;
+
+    // NOTE: we make an assumption here that our workingGeometry is a layer
+    // group with only one layer in it, so the iteration below will return a
+    // single layer. This assumption is enforced by the UI: it's only possible
```

Comment:
Unfortunately `editingLayerGroup` isn't an array, it's an object defined by Leaflet. However, we could build an array of layers as we iterate through `editingLayerGroup`, then just return the first element of this array.